File: problem_042.py
#!/usr/bin/env python
import re

print("Project Euler - Problem 42:")
print("In a list of two-thousand common English words, how many are triangle words?\n")

# Process input data
ReadFile = open("problem_042.txt")
data = ReadFile.readline()
data = re.compile('"').sub(r"", data)
data = data.split(",")
data = sorted(data)

def wordScore(n):
    value = 0
    for i in n:
        value += ord(i) - ord('@')
    return value

# Generate triangle numbers
triangles = [1]
for n in range(2, 100):
    triangles.append(triangles[-1]+n)

# Score words and see if they are triangle values
count = 0
for i in range(len(data)):
    score = wordScore(data[i])
    if score in triangles:
        count += 1

print(count)

File: problem_042.js
#!/usr/bin/node

console.log("Project Euler - Problem 42:");
console.log("In a list of two-thousand common English words, how many are triangle words?\n");

const fs = require('fs');
var buffer;

function score_word(word){
    score = 0;
    for (var c=0; c<word.length; c++)
	score += word.charCodeAt(c) - "A".charCodeAt(0) + 1;

    return score;
}

function count_triangle_words(buffer, triangles){
    var sum = 0;
    let words = buffer.split(",");

    for (var word of words)
	if (triangles.indexOf(score_word(word.replace(/"/g,""))) >= 0)
	    sum++;

    console.log(sum);
}

// Pre-computer triangle numbers
var triangles = [];
for (var i=1; i<100; i++)
    triangles.push(Math.floor(1/2 * i * (i + 1)));

// Read in text file
fs.readFile('problem_042.txt', 'utf8', (err, data) => {
    if (err) {
	console.error(err);
	return;
    }
    count_triangle_words(data, triangles);
})
